code/go/0chain.net/smartcontract/storagesc/readpool.go

```diff
@@ -193,159 +236,33 @@ func (ssc *StorageSmartContract) getReadPool(clientID datastore.Key,
 	input []byte, balances cstate.StateContextI) (resp string, err error) {
-
-	// configs
-
 	var conf *readPoolConfig
 	if conf, err = ssc.getReadPoolConfig(balances, true); err != nil {
 		return "", common.NewError("read_pool_lock_failed",
 			"can't get configs: "+err.Error())
 	}
 
-	var lr lockRequest
+	var lr readLockRequest
 	if err = lr.decode(input); err != nil {
 		return "", common.NewError("read_pool_lock_failed", err.Error())
 	}
 
-	if len(lr.TargetId) == 0 {
-		lr.TargetId = t.ClientID
-	}
-
-	var rp *readPool
-	if rp, err = ssc.getReadPool(lr.TargetId, balances); err != nil {
-		if err != util.ErrValueNotPresent {
-			return "", common.NewError("read_pool_lock_failed", err.Error())
-		}
-		rp = new(readPool)
-	}
-
-	if lr.AllocationID == "" {
-		return "", common.NewError("read_pool_lock_failed",
-			"missing allocation ID in request")
-	}
-
 	if t.Value < conf.MinLock {
 		return "", common.NewError("read_pool_lock_failed",
 			"insufficient amount to lock")
 	}
 
-	if lr.Duration < conf.MinLockPeriod {
-		return "", common.NewError("read_pool_lock_failed",
-			fmt.Sprintf("duration (%s) is shorter than min lock period (%s)",
-				lr.Duration.String(), conf.MinLockPeriod.String()))
-	}
-
-	if lr.Duration > conf.MaxLockPeriod {
-		return "", common.NewError("read_pool_lock_failed",
-			fmt.Sprintf("duration (%s) is longer than max lock period (%v)",
-				lr.Duration.String(), conf.MaxLockPeriod.String()))
-	}
-
-	// check client balance
-	if !lr.MintTokens {
-		if err = stakepool.CheckClientBalance(t, balances); err != nil {
-			return "", common.NewError("read_pool_lock_failed", err.Error())
-		}
-	}
-
-	// get the allocation object
-	var alloc *StorageAllocation
-	alloc, err = ssc.getAllocation(lr.AllocationID, balances)
-	if err != nil {
-		return "", common.NewError("read_pool_lock_failed",
-			"can't get allocation: "+err.Error())
-	}
-
-	var bps blobberPools
-
-	// lock for allocation -> blobber (particular blobber locking)
-	if lr.BlobberID != "" {
-		if _, ok := alloc.BlobberMap[lr.BlobberID]; !ok {
-			return "", common.NewError("read_pool_lock_failed",
-				fmt.Sprintf("no such blobber %s in allocation %s",
-					lr.BlobberID, lr.AllocationID))
-		}
-		bps = append(bps, &blobberPool{
-			Balance:   state.Balance(t.Value),
-			BlobberID: lr.BlobberID,
-		})
-	} else {
-		// divide depending read price range for all blobbers of the
-		// allocation
-		var total float64 // total read price
-		for _, b := range alloc.BlobberDetails {
-			total += float64(b.Terms.ReadPrice)
-		}
-		// calculate (divide)
-		for _, b := range alloc.BlobberDetails {
-			var ratio = float64(b.Terms.ReadPrice) / total
-			bps.add(&blobberPool{
-				Balance:   state.Balance(float64(t.Value) * ratio),
-				BlobberID: b.BlobberID,
-			})
-		}
-	}
-
-	// create and dig allocation pool
-
-	var ap allocationPool
-	ap.AllocationID = lr.AllocationID
-	ap.ExpireAt = t.CreationDate + toSeconds(lr.Duration)
-	ap.Blobbers = bps
-
-	if !lr.MintTokens {
```

Comment:
We need to put back the AddTransfer and AddMint calls back, otherwise no tokes will be moved.

Reply:
Can you please explain how MintTokens and AddTransfer work for the previous implementation so that I can work out how to implement them here?